detector/calorimeter/HCalTileBarrel_o1_v01_geo.cpp

```diff
@@ -223,12 +224,39 @@ static dd4hep::Ref_t createHCal(dd4hep::Detector& lcdd, xml_det_t xmlDet, dd4hep
   }
   
   // Place envelope (or barrel) volume
-  Volume motherVol = lcdd.pickMotherVolume(hCal);
+  Volume motherVol = lcdd.pickMotherVolume(caloDetElem);
   motherVol.setVisAttributes(lcdd.invisible());
-  PlacedVolume placedHCal = motherVol.placeVolume(envelopeVolume);
-  placedHCal.addPhysVolID("system", hCal.id());
-  hCal.setPlacement(placedHCal);
-  return hCal;
+  PlacedVolume envelopePhysVol = motherVol.placeVolume(envelopeVolume);
+  envelopePhysVol.addPhysVolID("system", xmlDet.id());
+  caloDetElem.setPlacement(envelopePhysVol);
+
+  
+  // Create caloData object
+  auto caloData = new dd4hep::rec::LayeredCalorimeterData;
+  caloData->layoutType = dd4hep::rec::LayeredCalorimeterData::BarrelLayout;
+  caloDetElem.addExtension<dd4hep::rec::LayeredCalorimeterData>(caloData);
+
+  caloData->extent[0] = sensitiveBarrelRmin;
+  caloData->extent[1] = sensitiveBarrelRmax; // or r_max ?
+  caloData->extent[2] = 0.;      // NN: for barrel detectors this is 0
+  caloData->extent[3] = dzDetector; 
+
+  dd4hep::rec::LayeredCalorimeterData::Layer caloLayer;
+
+  for (unsigned int idxLayer = 0; idxLayer < layerDepths.size(); ++idxLayer) {
+        const double difference_bet_r1r2 = layerDepths.at(idxLayer); 
+
+        caloLayer.distance                  = sensitiveBarrelRmin; // should this be always the radius of the first layer as it is now?  
```

Comment:
No, this should be the distance to the current layer

Reply:
Thanks for a clarification, will be corrected in the next commit